src/client/Inode.cc

```diff
@@ -702,6 +702,11 @@ bool Inode::delegations_broken(bool skip_read)
 
   if (skip_read) {
     Delegation& deleg = delegations.front();
+    if (deleg.get_type() == CEPH_FILE_MODE_WR && deleg.is_recalled()) {
```

Comment:
`deleg.is_recalled()` can only be true if the callback was invoked in https://github.com/ceph/ceph/blob/0df8f65e1b526c269854ac45a7be57bb35898221/src/client/Delegation.cc#L127-L130
 
Checking the nfs-ganesha code for the callback func https://github.com/nfs-ganesha/nfs-ganesha/blob/a26e2c33dbc4c005c65eefc666ba33ebac169dfd/src/FSAL/FSAL_CEPH/handle.c#L2636 and from the logs suhas shared, I think `up_async_delegrecall()` is called and didn't return any errno which means the deleg has been returned from the ganesha, now its just about marking it as recalled in libcephfs side and returning `true` for this case makes sense.